packages/headless/src/api/search/search-api-client.ts

```diff
@@ -91,6 +100,13 @@ export class SearchAPIClient {
       ...this.options,
     });
 
+    if (response instanceof Error) {
+      if (response instanceof DisconnectedError) {
+        return {error: buildDisconnectedError()};
+      }
+      throw response;
+    }
```

Comment:
Would it be possible for platform client to catch and return the `buildDisconnectedError` error object to avoid duplicating these `if` blocks in search-api client and case-assist api client?

Reply:
For the sake of keeping the return type of `PlatformClient.call` simpler, I decided to make it never throw and only return a response or an Error object.

To reduce repetition, I introduced a util.